amlip_cpp/src/cpp/types/AmlipIdDataType.cpp

```diff
@@ -0,0 +1,310 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file AmlipIdDataType.cpp
+ */
+
+#ifdef _WINID_SIZE
+// Remove linker warning LNK4221 on Visual Studio
+namespace {
+char dummy;
+}  // namespace
+#endif  // _WINID_SIZE
+
+#include <fastcdr/Cdr.h>
+
+#include <fastcdr/exceptions/BadParamException.h>
+using namespace eprosima::fastcdr::exception;
+
+#include <array>
+#include <utility>
+#include <random>
+#include <string>
+
+#include <types/AmlipIdDataType.hpp>
+
+namespace eprosima {
+namespace amlip {
+namespace types {
+
+const AmlipIdDataType AmlipIdDataType::UNDEFINED_ID_= AmlipIdDataType();
+
+AmlipIdDataType::AmlipIdDataType()
+{
+    memset(&name_, 0, (NAME_SIZE) * 1);
+    memset(&rand_id_, 0, (RAND_SIZE) * 1);
+}
+
+AmlipIdDataType::AmlipIdDataType(const std::string& name)
+{
+    name_ = str_name_to_array_(name);
+    rand_id_ = random_id_();
+}
+
+AmlipIdDataType::AmlipIdDataType(const char* name)
+    : AmlipIdDataType(std::string(name))
+{
+}
+
+AmlipIdDataType::AmlipIdDataType(
+    std::array<uint8_t, NAME_SIZE>& name,
+    std::array<uint8_t, RAND_SIZE>& rand_id)
+{
+    name_ = name;
+    rand_id_ = rand_id;
+}
+
+AmlipIdDataType::AmlipIdDataType(
+    std::array<uint8_t, NAME_SIZE>&& name,
+    std::array<uint8_t, RAND_SIZE>&& rand_id)
+{
+    name_ = std::move(name);
+    rand_id_ = std::move(rand_id);
+}
+
+AmlipIdDataType::~AmlipIdDataType()
+{
+}
+
+AmlipIdDataType::AmlipIdDataType(
+    const AmlipIdDataType& x)
+{
+    name_ = x.name_;
+    rand_id_ = x.rand_id_;
+}
+
+AmlipIdDataType::AmlipIdDataType(
+    AmlipIdDataType&& x)
+{
+    name_ = std::move(x.name_);
+    rand_id_ = std::move(x.rand_id_);
+}
+
+AmlipIdDataType& AmlipIdDataType::operator =(
+    const AmlipIdDataType& x)
+{
+    name_ = x.name_;
+    rand_id_ = x.rand_id_;
+
+    return *this;
+}
+
+AmlipIdDataType& AmlipIdDataType::operator =(
+    AmlipIdDataType&& x)
+{
+    name_ = std::move(x.name_);
+    rand_id_ = std::move(x.rand_id_);
+
+    return *this;
+}
+
+bool AmlipIdDataType::operator ==(
+    const AmlipIdDataType& x) const
+{
+    return (name_ == x.name_ && rand_id_ == x.rand_id_);
+}
+
+bool AmlipIdDataType::operator !=(
+    const AmlipIdDataType& x) const
+{
+    return !(*this == x);
+}
+
+std::string AmlipIdDataType::name() const
+{
+    std::string name = std::string(std::begin(name_), std::end(name_));
+    uint8_t null_idx = name.find_first_of('\0');
+    if (null_idx < name.size())
```

Comment:
NIT: This is maybe slightly faster and more understandable if you find the 0 in the array rather than the string, and then create the string.